File: Programmers/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.*;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {        
        Queue<Integer> bridge = new LinkedList<>();
        int time = 0;
        int curWeight = 0;
        
        for (int i = 0; i < bridge_length; i++) {
            bridge.offer(0);
        }
        
        int idx = 0;
        
        while (idx < truck_weights.length) {
            time++;
            
            curWeight -= bridge.poll();
            
            if (curWeight + truck_weights[idx] <= weight) {
                bridge.offer(truck_weights[idx]);
                curWeight += truck_weights[idx];
                idx++;
            } else {
                bridge.offer(0);
            }
        }
        
        return time + bridge_length;
    }
}

File: Programmers/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(int bridge_length, int weight, vector<int> truck_weights) {
    queue<int> bridge;
    int time = 0;
    int curWeight = 0;
    
    for (int i = 0; i < bridge_length; i++) bridge.push(0);
    
    int idx = 0;
    
    while (idx < truck_weights.size())
    {
        time++;
        
        curWeight -= bridge.front();
        bridge.pop();
        
        if (curWeight + truck_weights[idx] <= weight)
        {
            bridge.push(truck_weights[idx]);
            curWeight += truck_weights[idx];
            idx++;
        }
        else
        {
            bridge.push(0);
        }
    }
    
    return time + bridge_length;
}